account-compression/programs/account-compression/src/lib.rs

```diff
@@ -544,8 +564,11 @@ pub mod spl_account_compression {
         let merkle_tree_size = merkle_tree_get_size(&header)?;
         let (tree_bytes, canopy_bytes) = rest.split_at_mut(merkle_tree_size);
 
-        let id = ctx.accounts.merkle_tree.key();
-        merkle_tree_apply_fn_mut!(header, id, tree_bytes, prove_tree_is_empty,)?;
+        // Check if the tree is either empty or is batch initialized and not finalized yet.
+        if !header.get_is_batch_initialized() || !tree_bytes.iter().all(|&x| x == 0) {
```

Comment:
How does `!tree_bytes.iter().all(|&x| x == 0)` check that a batch initialized tree is not finalized yet?